langfuse-core/src/index.ts

```diff
@@ -1058,20 +1058,22 @@ export abstract class LangfuseCore extends LangfuseCoreStateless {
     }
 
     if (cachedPrompt.isExpired) {
-      return await this._fetchPromptAndUpdateCache({
-        name,
-        version,
-        label: options?.label,
-        cacheTtlSeconds: options?.cacheTtlSeconds,
-        maxRetries: options?.maxRetries,
-        fetchTimeout: options?.fetchTimeoutMs,
-      }).catch(() => {
-        console.warn(
-          `Returning expired prompt cache for '${this._getPromptCacheKey({ name, version, label: options?.label })}' due to fetch error`
-        );
+      // If the cache is not currently being refreshed, start refreshing it and register the promise in the cache
+      if (!this._promptCache.isRefreshing(cacheKey)) {
+        const refreshPromptPromise = this._fetchPromptAndUpdateCache({
+          name,
+          version,
+          label: options?.label,
+          cacheTtlSeconds: options?.cacheTtlSeconds,
+          maxRetries: options?.maxRetries,
+          fetchTimeout: options?.fetchTimeoutMs,
+        }).catch(() => {
+          console.warn(`Failed to refresh prompt cache '${cacheKey}'`);
```

Comment:
We should probably adjust this error message saying that we will serve the cached prompt until refreshing succeeded.